server/gitrest/packages/gitrest-base/src/utils/redisFs/redisFsManager.ts

```diff
@@ -258,10 +257,21 @@ export class RedisFs implements IFileSystemPromises {
 	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L152
 	 */
 	public async rmdir(folderpath: PathLike, options?: RmDirOptions): Promise<void> {
-		const keys_to_remove = await this.redisFs.keys_by_prefix(folderpath.toString());
+		const folderpathString = folderpath.toString();
+
+		const keysToRemove = await executeRedisFsApi(
+			async () => this.redisFsClient.keysByPrefix(folderpathString),
+			RedisFsApis.Rmdir,
+			RedisFSConstants.RedisFsApi,
+			this.redisFsConfig.enableRedisFsMetrics,
+			this.redisFsConfig.redisApiMetricsSamplingPeriod,
+			{
+				folderpathString,
+			},
+		);
 
-		keys_to_remove.forEach((key) => {
-			void this.redisFs.delete(key);
+		keysToRemove.forEach((key) => {
```

Comment:
Would it make sense to also wrap this in the `executeRedisFsAPI()` wrapper? For example:

```ts
const removeKeys = async (pathstr: string) => { 
            const keysToRemove = await this.redisFsClient.keysByPrefix(pathstr);
            keysToRemove.forEach((key) => {
			      void this.redisFsClient.delete(key);
		    });
};
const keysToRemove = await executeRedisFsApi(
			async () => removeKeys(folderpathString),
			RedisFsApis.Rmdir,
			RedisFSConstants.RedisFsApi,
			this.redisFsConfig.enableRedisFsMetrics,
			this.redisFsConfig.redisApiMetricsSamplingPeriod,
			{
				folderpathString,
			},
		);

		
```

Reply:
Also, why the fire-and-forget on the delete() call? (using void)
Could we instead use a Promise.all()?